src/pages/index.js

```diff
@@ -1,127 +1,248 @@
-const addFormValidator = new FormValidator(settings, addCardForm);
-editFormValidator.enableValidation();
-addFormValidator.enableValidation();
+enableValidation(settings);
 
 // Functions ===================================================================
 
-  function renderCard(card) {
-    const cardElement = new Card(
-      card,
-      (imageData) => {
-        cardImagePopup.open(imageData);
-      },
-      selectors.cardTemplate
-    );
-    return cardElement.getView();
-  }
- 
-
+function setSubmitButtonText(buttonElement, text) {
+  buttonElement.textContent = text;
+}
 
+function renderCard(card, userId) {
+  const cardElement = new Card({
+    data: card,
+    handleCardClick: (imageData) => {
+      console.log(card);
+      cardImagePopup.open(imageData);
+    },
+    cardSelector: selectors.cardTemplate,
+    handleDeleteClick: () => {
+      deleteCardPopup.setAction(() => {
+        setSubmitButtonText(deleteCardModalBtn, "Deleting...");
+        api
+          .deleteCard(card._id)
+          .then(() => {
+            cardElement._handleDelete();
+            deleteCardPopup.close();
+          })
+          .catch((err) => {
+            console.error(err.status)
+          })
+          .finally(() => {
+            setSubmitButtonText(deleteCardModalBtn, "Yes");
+          });
+      });
+      deleteCardPopup.open();
+    },
+    handleLikeClick: () => {
+      if (cardElement.isLiked()) {
+        api
+          .likeCountRemove(card._id)
+          .then((card) => {
+            cardElement.setLikes(card.likes);
+          })
+          .catch((err) => {
+            console.error(err.status);
+          });
+      } else {
+        api.likeCountAdd(card._id).then((card) => {
+          cardElement.setLikes(card.likes);
+        });
+      }
+    },
+    userId: userId,
+  });
 
+  return cardElement.getView();
+}
 
-//        Modal Image       ===============================================
+//   Card Image Modal  =================================================
 
 const cardImagePopup = new PopupWithImage(selectors.cardImageModal);
 
 cardImagePopup.setEventListeners();
 
-// Card Section ================================================================
+// Delete Card Modal  ====================================================
+
+const deleteCardPopup = new PopupWithConfirmation(selectors.deleteCardModal);
+deleteCardPopup.setEventListeners();
+
+// Add Card  ===============================================================
+
+const addCardPopup = new PopupWithForm(selectors.addCardModal, (data) => {
+  setSubmitButtonText(addCardSubmitBtn, "Creating...");
+
+  api
+    .addCard(data)
+    .then((data) => {
+      const newCard = renderCard(data, userId);
+      sectionInstance.addItem(newCard);
+    })
+    .catch((err) => {
+      console.error(err.status);
+    })
+    .then(() => {
+      addCardPopup.close();
+      formValidators["add-card-form"].disableButton()
```

Comment:
`.disableButton()` should be called when you **open** modals because when you close them you reset the form according to the task